gc/base/standard/Scavenger.cpp

```diff
@@ -3568,8 +3568,8 @@ MM_Scavenger::completeBackOut(MM_EnvironmentStandard *env)
 	OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
 #endif /* OMR_SCAVENGER_TRACE_BACKOUT */
 
-	/* Ensure we've pushed all references from buffers out to the lists */
-	_cli->scavenger_flushReferenceObjects(env);
+	/* Ensure we've pushed all references from buffers out to the lists and flushed RS fragments*/
+	flushBuffersForGetNextScanCache(env);
```

Comment:
I believe this problem exists in legacy IBM R2.7 and R2.8 VM releases, too. Ever since flush was moved from the end of scan loop in the middle of getNextScanCache()

When there is an aborted scavenge, threads rush to terminate the scan loop and may exit it without doing the intermediate flush.

We actually have this assert at the end of scan loop (which means we are aware that the count may be unflushed if there is a backout):
	Assert_MM_true(backOutRaisedThisScanCycle || (0 == env->_scavengerRememberedSet.count));

Reply:
Thanks for the clarification!